src/core/metadata/qgslayermetadatavalidator.h

```diff
@@ -60,27 +60,23 @@ class CORE_EXPORT QgsAbstractMetadataBaseValidator
         //! Metadata section which failed the validation
         QString section;
 
-        // TODO QGIS 4.0 - fix this
-
-#ifdef SIP_RUN
-        SIP_PROPERTY( name = identifier, get = _identifier, set = _setIdentifier )
-#endif
+        // TODO QGIS 4.0 - remove compatibility code
```

Comment:
Why are we removing the property ? Some plugin code could miss it, no?

Reply:
@troopa81 It's still there, just moved to a python only implementation instead of sip based 

I prefer this approach anyway, it keeps the headers free from the noise